databricks-sdk-java/src/main/java/com/databricks/sdk/core/utils/ProxyUtils.java

```diff
@@ -142,4 +158,29 @@ public static void setupBasicProxyAuth(
         .setDefaultCredentialsProvider(credsProvider)
         .setProxyAuthenticationStrategy(new ProxyAuthenticationStrategy());
   }
+
+  static class CustomRoutePlanner implements HttpRoutePlanner {
```

Comment:
Can we move this to its own file if it doesn't have any dependencies in the parent class?